apollo-core/src/main/java/com/ctrip/framework/apollo/core/enums/ConfigSyncType.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2022 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.core.enums;
+
+import com.ctrip.framework.apollo.core.utils.StringUtils;
+
+import java.util.stream.Stream;
+
+/**
+ * This enum represents all the possible Configuration sync  from apollo-config
+ *
+ * @author jason
+ */
+public enum ConfigSyncType {
+  FULLSYNC("FullSync"), INCREMENTALSYNC("IncrementalSync");
+
+  private final String value;
+
+  ConfigSyncType(String value) {
+    this.value = value;
+  }
+
+
+
+  /**
+   * Transforms a given string to its matching {@link ConfigSyncType}.
+   *
+   * @param value the string that matches
+   * @return the matching {@link ConfigSyncType}
+   * @throws IllegalArgumentException in case the <code>value</code> is empty or there is no
+   *                                  matching {@link ConfigSyncType}
+   */
+  public static ConfigSyncType fromString(String value) {
+    if (StringUtils.isEmpty(value)) {
+         return FULLSYNC;
+    }
+    for (ConfigSyncType type : values()) {
+            if (type.value.equals(value)) {
+                return type;
+             }
+         }
+       throw new IllegalArgumentException("Invalid ConfigSyncType: " + value);
```

Comment:
Adding a sync mode like `UNKNOWN` helps ensure things stay compatible for the future. You may reference the implementation of `com.ctrip.framework.apollo.core.enums.EnvUtils#transformEnv`.